ppfleetx/core/engine/eager_engine.py

```diff
@@ -176,15 +177,21 @@ def configure_optimizers(self):
 
         self._use_recompute = configs['Model']['use_recompute']
 
-        if self._use_pure_fp16:
-            if mode == 'train':
+        if self._amp_enable:
+            if mode == 'train' and self._amp_dtype == "float16":
                 self._scaler = paddle.amp.GradScaler(
                     init_loss_scaling=self._scale_loss)
+            else:  # bfloat16
+                self._scaler = paddle.amp.GradScaler(
+                    init_loss_scaling=1, use_dynamic_loss_scaling=False)
 
             # Save dtype is the same as model dtype. Also can set save_dtype='float32' when
             # training with pure fp16 strategy, but will cause the rise of memory.
-            self._module.model = paddle.amp.decorate(
-                models=self._module.model, level='O2')
+            if self._amp_level == "O2":
```

Comment:
Does the setting O2 and bfloat16 turn on the training of bfloat16? what's the meaning of the setting O1 and bfloat16?